### PR TITLE
Create a valid no-op dylib for the iOS simulator.

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

**Comment by chinmaygarde** [comment]
cc @abarth @tvolkert

File: sky/build/sdk_xcode_harness/Tools/common/BuildFlutterApp [hunk 1]
```diff
@@ -64,7 +64,7 @@ BuildApp() {
       exit -1
     fi
   else
-    RunCommand touch ${derived_dir}/app.so
+    RunCommand eval "$(echo \"static const int Moo = 88;\" | xcrun clang -x c --shared -o ${derived_dir}/app.so -)"
```
Comment (abarth) [inline sky/build/sdk_xcode_harness/Tools/common/BuildFlutterApp]: 42

**Comment by abarth** [comment]
LGTM

**Comment by tvolkert** [comment]
lgtm

**Pull request merged**